### PR TITLE
Plugin to change previous/next post to reflect section indices instead of global index

### PR DESCRIPTION
**Pull request by wirew0rm**
Plugin to solve getnikola/nikola#2625.

Subscribe to the 'taxonomies_classified' signal and upon receiving it update next_post and previous_post for every post to its successor/predecessor in it's corresponding section.

### CODE REVIEW

File: v7/section_prevnext/section_prevnext.plugin [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Core]
+name = section_prevnext
+module = section_prevnext
+
+[Documentation]
+author = Alexander Krimm
+version = 0.8
+website = https://getnikola.com/
```
Comment (Kwpolska) [inline v7/section_prevnext/section_prevnext.plugin]: Perhaps use your own website here?

File: v7/section_prevnext/section_prevnext.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © 2012-2017 Roberto Alsina and others.
```
Comment (Kwpolska) [inline v7/section_prevnext/section_prevnext.py]: 2017 only, and your name, goes here

File: v7/section_prevnext/section_prevnext.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+# PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+# OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""Render the taxonomy overviews, classification pages and feeds."""
```
Comment (Kwpolska) [inline v7/section_prevnext/section_prevnext.py]: This docstring, and the one in SectionNav, should be more descriptive.

File: v7/section_prevnext/section_prevnext.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    def _set_navlinks(self, site):
+        # Needed to avoid strange errors during tests
+        if site is not self.site:
+            return
+        # Update prev_post and next_post
+        for lang, langposts in site.posts_per_classification['section_index'].items():
+            for section, sectionposts in langposts.items():
+                for i, p in enumerate(sectionposts[1:]):
+                    print(p.title)
```
Comment (Kwpolska) [inline v7/section_prevnext/section_prevnext.py]: Remove debug print statement here

File: v7/section_prevnext/section_prevnext.plugin (hunk 1 above)
Comment (wirew0rm) [inline v7/section_prevnext/section_prevnext.plugin]: Unfortunately it is not up yet, i've updated it to point to the plugin page itself.

File: v7/section_prevnext/section_prevnext.plugin [hunk 5]
```diff
@@ -4,9 +4,10 @@ module = section_prevnext
 
 [Documentation]
 author = Alexander Krimm
-version = 0.8
-website = https://getnikola.com/
+version = 0.1
+website = https://plugins.getnikola.com/#section_prevnext
```
Comment (Kwpolska) [inline v7/section_prevnext/section_prevnext.plugin]: that isn’t going to be the URL anymore. You can use your own homepage here, or whatever else.

**Pull request merged**